### PR TITLE
Add qjit-ted "Symmetry-invariant quantum machine learning force fields" demo

### PR DESCRIPTION
**Pull request by paul0403**
**Context:**
As part of Catalyst's work on [identifying 10 demos to compile with catalyst](https://app.shortcut.com/xanaduai/epic/65250/p1-discovery-10-demos-are-identified-for-catalyst-conversion-2-implemented?group_by=none&vc_group_by=day&ct_workflow=all&cf_workflow=500000005), we convert ["Symmetry-invariant quantum machine learning force fields"](https://pennylane.ai/qml/demos/tutorial_eqnn_force_field), a machine learning workflow that calls a training step function repeatedly and thus can have significant performance boosts with catalyst compilation.

**Description of the Change:**
Convert the demo to use qjit instead of jax.jit

**Benefits:**
Demonstration of qjit advantage

**Possible Drawbacks:**
Because lightning does not support differentiation through `QubitUnitary`, we have to resort to finite difference method for gradients, which brings a big performance degradation.

**Related GitHub Issues:**
[sc-72938]

### CODE REVIEW

**Comment by paul0403** [comment]
To me here are the possible paths forward:
(a) make lightning work with differentiating through qubit unitary (will take a bit of time)
(b) decompose qubit unitary into "standard" gateset, although unsure whether this decomposition itself can be jitted or not 
(c) decrease the batch size and the number of traning steps so the demo can finish running in a reasonable amount of time

Personally, (c) makes the most sense for me at this stage.

File: demos/qml/tutorial_eqnn_force_field.py [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+net_params = {"params": {"weights": weights, "alphas": alphas, "epsilon": epsilon}}
+opt_state = opt_init(net_params)
+running_loss = []
+#################################
+# We train our VQLM using stochastic gradient descent.
+
+
+num_batches = 200 # 5000  # number of optimization steps
+batch_size = 5 # 256  # number of training data per batch
+
```
Comment (paul0403) [inline demos/qml/tutorial_eqnn_force_field.py:490]: This smaller-sized run can finish in a reasonable amount of time (around 5 minutes)

**Comment by josh146** [comment]
@paul0403 I recommend not adding the demo here -- I am in the process of moving all demos _out_ of the Catalyst docs and into pennylane.ai/qml, to ensure that:

- We have them in a single place to ensure that they stay up to date
- In the QML repo, they are tested for accuracy regularly through integration tests

**Comment by paul0403** [comment]
> I am in the process of moving all demos _out_ of the Catalyst docs and into pennylane.ai/qml

I was not aware of this! Thanks for letting me know!
